histFactory/CMakeLists.txt

```diff
@@ -41,7 +41,7 @@ else()
 endif()
 
 set(Boost_NO_BOOST_CMAKE ON)
-find_package(Boost REQUIRED)
+find_package(Boost 1.57.0 REQUIRED COMPONENTS system filesystem)
```

Comment:
Without this requirement cmake would fetch the system and filesystem libraries from the system instead of the BOOST_ROOT directory... I spent some time fiddling around but in the end that was the only thing that worked!